qoffer-domain/src/main/groovy/life/qbic/portal/portlet/offers/create/CreateOffer.groovy

```diff
@@ -8,5 +13,32 @@ package life.qbic.portal.portlet.offers.create
  * @since: 1.0.0
  * @author: Tobias Koch
  */
-class CreateOffer {
+class CreateOffer implements CreateOfferInput{
+
+    private CreateOfferDataSource gateway
+    private CreateOfferOutput output
+
+    CreateOffer(CreateOfferDataSource gateway, CreateOfferOutput output){
+        this.gateway = gateway
+        this.output = output
+    }
+
+    @Override
+    void createOffer(String tomatoId) {
+
+    }
+
+    @Override
+    void createNewOffer(Quotation quotation) {
+        gateway.createOffer(quotation.identifier,quotation.projectTitle,quotation.projectDescription, quotation.customer,quotation.projectManager,quotation.items)
+    }
+
```

Comment:
I would see it in the responsibility of the use case to create the id and thus we would need to create another constructor for the quotation or a method to add the allows to add an id to the current quotation